.github/workflows/docker-cloudstack-simulator.yml

```diff
@@ -0,0 +1,61 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Docker Image Build
+
+on:
+  push:
+    branches:
+      - main
+    tags:        
+      - 4.*
+      - 5.*
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Login to Docker Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ secrets.DOCKER_REGISTRY }}
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
```

Comment:
Are these available to the upstream repo @weizhouapache or are we going to ask ASF infra to add those?

Reply:
@rohityadavcloud 
We need to ask ASF infra to add these secrets to Github project settings.

Before that, I think we need to discuss 
- which repository is official (https://hub.docker.com/ or https://ghcr.io/)
- who have access 
- How many docker images (for commits in main branch) need to be kept.
- other questions

Reply:
Can you do and log a request with ASF-infra (try slack and JIRA) that before we merge this, otherwise this job will fail (to publish/push).